src/main/scala/cromwell/engine/SymbolTable.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.engine
+
+import cromwell.binding.types.WdlType
+import cromwell.binding.{FullyQualifiedName, Call, WdlBinding, WdlValue}
+import scala.collection.mutable
+
+class SymbolTableEntry(val scope: String, val name: String, val iteration: Int, var wdlValue: Option[WdlValue], val wdlType: WdlType, input: Boolean) {
```

Comment:
The table itself must be mutable to be able to add new entries dynamically (e.g. dynamic scatter), so I basically punted on immutability in this corner of the system.  :smile: